netman/adapters/switches/juniper/base.py

```diff
@@ -1101,20 +1124,24 @@ def _compute_edge_state_modifications(interface_id, edge, config):
     modifications = []
     rstp_node = first(config.xpath("data/configuration/protocols/rstp/interface/name[text()=\"{0:s}\"]/.."
                                    .format(interface_id)))
+
+    edge_node = None
+    no_root_port_node = None
     if rstp_node is not None:
         edge_node = first(rstp_node.xpath("edge"))
         no_root_port_node = first(rstp_node.xpath("no-root-port"))
 
-        if edge is True:
-            if edge_node is None:
-                modifications.append("<edge />")
-            if no_root_port_node is None:
-                modifications.append("<no-root-port />")
-        elif edge is False:
-            if edge_node is not None:
-                modifications.append("<edge operation=\"delete\" />")
-            if no_root_port_node is not None:
-                modifications.append("<no-root-port operation=\"delete\" />")
+    if edge is True:
```

Comment:
OK, new code creates the rstp_node if it doesnt exist, whereas before setting edge would just noop. Nice catch!